lxd/storage_pools.go

```diff
@@ -930,6 +921,8 @@ func storagePoolDelete(d *Daemon, r *http.Request) response.Response {
 
 	// Only perform the deletion of remote volumes on the server handling the request.
 	if !clusterNotification || !pool.Driver().Info().Remote {
+		projectName := projectParam(r)
```

Comment:
@stgraber I wanted to draw your attention to this. 

For `storagePoolPut` and `storagePoolDelete` I am now using `project.Default` for the lifecycle and `SyncResponseLocation` URL. My thinking here is that even if the request has come from a user in a non-default project, the resource itself (the storage pool) always belongs in project.Default.

However for `storagePoolDelete` this function also removes images on the pool before deleting it, and for that it has historically used the requested project to only delete images belonging to the current project. However if there are images belonging to other projects on the pool, the images will be left and the storage pool deletion will fail.

Should we change this to delete all images across all projects when deleting a storage pool so it doesn't fail?

Reply:
Yeah, we should delete images from all projects. If you're privileged enough to be allowed to delete a storage pool, you're also privileged enough to delete those images anyway.

Reply:
Great, thats what I hoped.

I'll follow that up as a separate PR.